internal/fixed_manifest_image.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: BSD-2-Clause
+
+package internal
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"hash"
+	"strings"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+)
+
+type fixedManifestImage struct {
+	v1.Image
+}
+
+// FixedManifestImage returns a image that produces a manifest that conforms to
+// the canonical form found in registries:
+// - JSON pretty print of the manifest with 3 spaces indentation.
```

Comment:
This might not be a canonical form BTW, it might just be a formatting that was passing from one registry to the next, **seems the target registry of a push just takes the manifest given as is**. So this format I thought to be canonical or standard here might be only due to some weirderness doen by the original tool that crafted the original manifest before pushing it to any registry.

That is a compelling argument not to use this approach and just record the original manifest in the bundle so we can reconstruct the original manifest/digest pair at load time.